### PR TITLE
issue #298 HTTP Request ContentType urlencoded not supported

### PR DESCRIPTION
**Pull request by nsiitk**
used http header content-type to overrider application/json 
   	 "http_request": {
	      	  	"uri": "http://localhost:8000/oauth/token",
	          	"method": "POST",        	"body":"username=${workflow.input.user}&password=${workflow.input.pwd}&client_id=trusted-app&grant_type=password",
	        	"contentType": "application/x-www-form-urlencoded",
	        	"accept":"application/json",
				"headers":{"content-type":"application/x-www-form-urlencoded"}
	        	}
	      	}

### CODE REVIEW

**Comment by v1r3n** [comment]
@nsiitk can you send the PR against the `dev` branch?

**Comment by nsiitk** [comment]
@v1r3n changed PR to dev. Don't understand why 2 test cases failing as my change don't impact those test cases.

File: contribs/src/main/java/com/netflix/conductor/contribs/http/HttpTask.java [hunk 1]
```diff
@@ -159,13 +159,14 @@ protected HttpResponse httpCall(Input input) throws Exception {
 			OAuthSecrets secrets = new OAuthSecrets().consumerSecret(input.oauthConsumerSecret);
 			client.addFilter(new OAuthClientFilter(client.getProviders(), params, secrets));
 		}
-
-		Builder builder = client.resource(input.uri).type(MediaType.APPLICATION_JSON);
+		final String contentType = "content-type";
```
Comment (v1r3n) [inline contribs/src/main/java/com/netflix/conductor/contribs/http/HttpTask.java]: What about different variations?  e.g. `Content-Type` ?

Comment (nsiitk) [reply contribs/src/main/java/com/netflix/conductor/contribs/http/HttpTask.java]: http headers are case insensitive so header hashmap should have all keys in same case.
If I change the setter of header in Input class all header will be lower case.
public void setHeaders(Map<String, Object> headers) {
			 for (String key : headers.keySet()) {
		            this.headers.put(key.toLowerCase(),headers.get(key));
		        }
		}

Comment (picaron) [reply contribs/src/main/java/com/netflix/conductor/contribs/http/HttpTask.java]: @nsiitk @v1r3n according to the Conductor [docs](https://netflix.github.io/conductor/metadata/systask/#http), there is a distinct parameter `contentType` that should be used for setting the content type of the request. 

I created a similar PR to use that parameter, see PR #391

Comment (v1r3n) [reply contribs/src/main/java/com/netflix/conductor/contribs/http/HttpTask.java]: @nsiitk as @picaron mentioned, `contentType` is a distinct parameter and should be used to override the contentType.

**Pull request closed without merging**